product/views.py

```diff
@@ -66,3 +68,40 @@ def get(self, request):
             return JsonResponse({'message':'KEY_ERROR'}, status=400)
         except Category.DoesNotExist:
             return JsonResponse({'message':'CATEGORY_DOES_NOT_EXIST'}, status=404)
+
+
+# 상품 상세 보기
+class ProductView(View):
+    def get(self, request, product_id):
+        try:
+            product = product_id
+
+            product = Product.objects.filter(id=product_id).prefetch_related('option', 'image_url')
+            product = product[0]
+
+            image_list = [{
+                'image_url' : product.image_url.all()[i].image_url
+            } for i in range(len(product.image_url.all())
+            )]
+
+            option_list = [{
+                'option_name'  : product.option.all()[i].name,
+                'option_price' : product.option.all()[i].price
+            } for i in range(len(product.option.all())
+            )]
+
+            product_dict = {
+                'product_id'    : product_id,
+                'product_name'  : product.name,
+                'product_price' : product.price,
+                'description'   : product.description,
+                'image_list'    : image_list,
+                'option_list'   : option_list
+            }
```

Comment:
이렇게 처리하면 코드가 많이 수동적인것 같아요.
```suggestion
            product = Product.objects.filter(id=product_id).prefetch_related('option', 'image_url')[0]           
            
            product_dict = {
                'product_id'    : product.id,
                'product_name'  : product.name,
                'product_price' : product.price,
                'description'   : product.description,
                'image_list'    : product.image_url.values_list('image_url', flat=True),
                'option_list'   : [
                    {
                        'option_name' : option.name,
                        'option_price' : option.price
                     } for option in product.option.all()
                 ]
            }
```
이렇게 해보면 어떨까요?

Reply:
멋져요!!!!!!:+1:

Reply:
지훈님 values_list로 받은 값이 쿼리셋 객체라 json serialize오류가 나길래 for문으로 빼서 넣었어요! 혹시 더 좋은 방법이 있을까요?

Reply:
음 아마 저코드에서는 그냥 저 values_list찍는 코드에 [ ]로  감싸줬으면 되지 않았을까 생각됩니다~!

Reply:
한번 해보시고 잘되면 이걸로 해서 제출해주세여~!

Reply:
제출했습니다~!~!!! :+1: